src/OrchardCore/OrchardCore.Users.Core/Extensions/ExternalLoginInfoExtensions.cs

```diff
@@ -0,0 +1,9 @@
+using System.Security.Claims;
+
+namespace Microsoft.AspNetCore.Identity;
+
+public static class ExternalLoginInfoExtensions
+{
+    public static string GetEmail(this ExternalLoginInfo info)
+        => info.Principal.FindFirstValue(ClaimTypes.Email) ?? info.Principal.FindFirstValue("email");

```

Comment:
Is "email" set somewhere?

Reply:
do you want the extension method return a nullable?

Reply:
I mean where the value "email" come from?

Reply:
external login data ... if the login provider have the user email

Reply:
AFAIK there are built-in claims, unless you add a custom one, that's why I'm asking about "email"

Reply:
Ah, sorry. Now I have understand. I don't know ... I have see the check done in this way and reported it as-is.

https://github.com/OrchardCMS/OrchardCore/blob/9d79ab57351a0703e72c5a9278d6d2739de440a1/src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs#L404-L406

I can remove "email" if you believe that is wrong.

Reply:
I might look the original author, it's not your fault :)

Reply:
I think this is reading the `email` claim from the external principal.

Reply:
Which one do you mean? I didn't see any.

Reply:
Your ask

> I mean where the value "email" come from?

I think it comes from the external provider via scope. So it would be a claim in the principal

Reply:
I'm not sure why the original author added it!!

Let us not prevent this PR from merging, but we need to figure out this

Reply:
Please approve and/or merge then